CONTRIBUTING.md

```diff
@@ -289,6 +299,32 @@ do not decrease overall code coverage of the codebase - this is aligned with our
 goal to increase coverage over time. Keep track of execution time for your unit
 tests and try to keep them as short as possible.
 
+#### Library Recommendations 
+
+To keep testing practices consistent across the project, it is advised to use this libraries under
+these circumstances:
+
+- For assertions to validate expecations, use `"github.com/stretchr/testify/assert"`
+- For assertions that are required to continue the test, use `"github.com/stretchr/testify/require"`
+- For mocking external resources, use `"github.com/stretchr/testify/mock"`
+- For validating HTTP traffic interactions, `"net/http/httptest"`
+- For generating test data, use `???`
+
+### Integration Testing
+
+Integration testing is encourage throughout the project, docker images can be used in order to facility
+a local version. The recommend library to control docker images throughout testing is `???`.
+In the absence of locally running version, mocking should be used to assert expectations.
+
+### Using CGO
+
+Using CGO should be avoided due to the lack of portability and complexity 
```

Comment:
```suggestion
Using CGO is prohibited due to the lack of portability and complexity 
```

Reply:
As far as I know we don't allow CGO at all currently. Not sure if there is a need to carve out exceptions. If the component is never going to be included in the build here it is best to make it clear upfront.

Reply:
Is that only true for the collector? or does that include the contrib?

Reply:
We use cgo optionally on the hostmetricsreceiver macOS implementation (see [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/hostmetricsreceiver#notes)), I think that is the only place we use it today

Reply:
Alright, sounds good to me :)